spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkScanBuilder.java

```diff
@@ -213,40 +215,53 @@ public Scan build() {
 
     Schema expectedSchema = schemaWithMetadataColumns();
 
-    TableScan scan =
-        table
-            .newScan()
-            .caseSensitive(caseSensitive)
-            .filter(filterExpression())
-            .project(expectedSchema);
+    if (startSnapshotId == null) {
+      BatchScan scan =
+          table
+              .newBatchScan()
+              .caseSensitive(caseSensitive)
+              .filter(filterExpression())
+              .project(expectedSchema);
 
-    if (snapshotId != null) {
-      scan = scan.useSnapshot(snapshotId);
-    }
+      if (snapshotId != null) {
+        scan = scan.useSnapshot(snapshotId);
+      }
 
-    if (asOfTimestamp != null) {
-      scan = scan.asOfTime(asOfTimestamp);
-    }
+      if (asOfTimestamp != null) {
+        scan = scan.asOfTime(asOfTimestamp);
+      }
 
-    if (branch != null) {
-      scan = scan.useRef(branch);
-    }
+      if (branch != null) {
+        scan = scan.useRef(branch);
+      }
 
-    if (tag != null) {
-      scan = scan.useRef(tag);
-    }
+      if (tag != null) {
+        scan = scan.useRef(tag);
+      }
+
+      scan = configureSplitPlanning(scan);
+
+      return new SparkBatchQueryScan(
+          spark, table, scan, readConf, expectedSchema, filterExpressions);
+
+    } else {
+      IncrementalAppendScan scan =
+          table
+              .newIncrementalAppendScan()
+              .fromSnapshotExclusive(startSnapshotId)
```

Comment:
I believe the `startSnapshotId` option is always exclusive. 

```
// Start snapshot ID used in incremental scans (exclusive)
public static final String START_SNAPSHOT_ID = "start-snapshot-id";
```

We may want to add a new boolean option to make it configurable.